pkgs/applications/office/ticktick/default.nix

```diff
@@ -0,0 +1,79 @@
+{ lib
+, fetchurl
+, stdenv
+, wrapGAppsHook
+, dpkg
+, autoPatchelfHook
+, glibc
+, gcc-unwrapped
+, nss
+, libdrm
+, mesa
+, alsa-lib
+, xdg-utils
+, systemd
+}:
+stdenv.mkDerivation (final: {
+  pname = "ticktick";
+  version = "1.0.80";
+
+  src = fetchurl {
+    url = "https://d2atcrkye2ik4e.cloudfront.net/download/linux/linux_deb_x64/${final.pname}-${final.version}-amd64.deb";
+    sha256 = "1iyxn9rdv6p8d36swqpxb346arblk00yhv9z2qf6i6r2a4sbrbqh";
+  };
+
+  system = "x86_64-linux";
+
+  nativeBuildInputs = [
+    wrapGAppsHook
+    autoPatchelfHook
+    dpkg
+  ];
+
+  buildInputs = [
+    nss
+    glibc
+    libdrm
+    gcc-unwrapped
+    mesa
+    alsa-lib
+    xdg-utils
+  ];
+
+  # Needed to make the process get past zygote_linux fork()'ing
+  runtimeDependencies = [
+    systemd
+  ];
+
+  unpackPhase = ''
+    runHook preUnpack
+
+    mkdir -p "$out/share" "$out/opt/${final.pname}" "$out/bin"
+    dpkg-deb --fsys-tarfile "$src" | tar --extract --directory="$out"
+
+    runHook postUnpack
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    cp -av $out/opt/TickTick/* $out/opt/${final.pname}
+    cp -av $out/usr/share/* $out/share
+    rm -rf $out/usr $out/opt/TickTick
+    ln -sf "$out/opt/${final.pname}/${final.pname}" "$out/bin/${final.pname}"
+    sed -i \
```

Comment:
I've just realised you could use `substituteInPlace` instead of `sed` here.
https://nixos.org/manual/nixpkgs/stable/#fun-substitute
```
substituteInPlace "$out/share/applications/${final.pname}.desktop" \
  --replace "Exec=/opt/TickTick/ticktick" "Exec=$out/bin/${final.pname}"
```
(didn't test this code)

Reply:
@TheBrainScrambler What's the motivation for using `substituteInPlace` over `sed`? For future reference mainly, that `sed` should work on every build.

Reply:
It's not strictly needed, but is more idiomatic and thus easier to read.
https://discourse.nixos.org/t/when-to-use-substituteinplace-functions-vs-a-patch/11073/10